### DISCUSSION TITLE
Ensuring repeatable global histogram results

### DISCUSSION THREAD

**Discussion by nullpainter** [SixLabors/ImageSharp#1413] [Q&A] [answered]
I've noticed that output images are very subtly different after histogram equalisation is applied with the default global options. 

As a pretty basic test, the following code:

```c#
for (var i = 0; i < 5; i++)
{
    using var image = await Image.LoadAsync(@"c:\temp\input.jpg");
    image.Mutate(c => c.HistogramEqualization());
    await image.SaveAsync($@"c:\temp\output_{i}.png");
}
```

produces files all with slightly different sizes:

```
06/11/2020  05:48 pm         3,445,866 0.png
06/11/2020  05:48 pm         3,429,505 1.png
06/11/2020  05:48 pm         3,449,050 2.png
06/11/2020  05:48 pm         3,434,962 3.png
06/11/2020  05:48 pm         3,434,531 4.png
```

I haven't looked into the code too much, but I'm assuming there's some sort of random sampling going on. Is there a way to ensure repeated applications of `HistogramEqualization` produce consistent results?

The background is that I'm applying a CLUT to an intensity range of a greyscale image, after histogram equalisation has been applied. The subtly different results between runs is resulting in noticeable differences in the CLUT-enhanced output and it would be good to avoid this if possible.

**Comment by antonfirsov** [comment, marked as answer]
@nullpainter this is weird, I can't see any randomness in the processor implementations, the outputs should be deterministic.

Can you double check this with an image diff tool, and if you can prove there is a pixel difference:
open an issue with template filled + images attached (or even better a repro code doing a pixel comparison)?

**Reply by JimBobSquarePants** [reply]
I managed to recreate this so @nullpainter please raise an issue.

**Reply by nullpainter** [reply]
Will do. Would you like me to make a little repro project?

**Reply by nullpainter** [reply]
I'll make one anyway, just for fun.

**Reply by nullpainter** [reply]
Done (#1416). Thanks!

**Reply by brianpopow** [reply]
The issue is now fixed with the release of [v1.0.2](https://github.com/SixLabors/ImageSharp/releases/tag/v1.0.2)
